perception/mil_vision/mil_vision_tools/rect_finder.py

```diff
@@ -0,0 +1,179 @@
+
+        If the corners are found, it returns these 4 points sorted as described
+        in sort_corners. These corners can be used for get_pose_2D and get_pose_3D
+
+        If a 4 sided polygon cannot be approximated, returns None.
+
+        @param epsilon_factor passed to OpenCV approx polygon function. May have to be tuned for your setup.
+        @param debug_image if not None, will draw corners with text for indexes onto image
+        '''
+        epsilon = epsilon_factor*cv2.arcLength(contour,True)
+        polygon = cv2.approxPolyDP(contour,epsilon,True)
+        if len(polygon) != 4:
+            return None
+        return self.sort_corners(polygon, debug_image=debug_image)
+
+    def get_pose_3D(self, corners, intrinsics=None, dist_coeffs=None, cam=None, rectified=False):
+        '''
+        Uses the model of the object, the corosponding pixels in the image, and camera
+        intrinsics to estimate a 3D pose of the object.
+
+        @param corners 4x2 numpy array from get_corners representing the 4 sorted corners in the image
+
+        One of the two must be set:
+        @param cam PinholeCameraModel object from ImageGeometry package
+        @param rectified if cam is set, set True if corners were found in an already rectified image (image_rect_color topic)
+
+        Or:
+        @param instrinsics camera intrinisic matrix as numpy array
+        @param dist_coeffs camera distortion coefficients as numpy array
+
+        @return tuple (tvec, rvec) representing the translation and rotation vector
+        between the camera and the object in meters/radians. Use cv2.Rodrigues to convert
+        rvec to a 3x3 rotation matrix.
+        '''
+        corners = np.array(corners,dtype=np.float)
+        # Use camera intrinsics and knowledge of marker's real demensions to get a pose estimate in camera frame
+        if cam is not None:
+            intrinsics = cam.intrinsicMatrix()
+            if rectified:
+                dist_coeffs = np.zeros((5,1))
+            else:
+                dist_coeffs = cam.distortionCoeffs()
+        assert intrinsics is not None
+        assert dist_coeffs is not None
+        _, rvec, tvec =  cv2.solvePnP(self.model_3D, corners, intrinsics, dist_coeffs)
+        return (tvec, rvec)
+
+    def get_pose_2D(self, corners):
+        '''
+        Finds the 2D center of the rectangle and a unit direction vector along the length
+        of the rectangle in pixels
+
+        @param corners 4x2 numpy array from get_corners representing the 4 sorted corners in the image
+
+        @return a tuple (center, vecter)
+        '''
+        top_center = (corners[1]+corners[0])/2.0
+        bot_center = (corners[2]+corners[3])/2.0
+        vector = top_center - bot_center
+        vector = vector / np.linalg.norm(vector)
+        center = bot_center + (top_center - bot_center)/2.0
+        vector = top_center - bot_center
+        vector = vector/np.linalg.norm(vector)
```

Comment:
I corrected all the issues outputted by flake8 in all the files changed in this PR